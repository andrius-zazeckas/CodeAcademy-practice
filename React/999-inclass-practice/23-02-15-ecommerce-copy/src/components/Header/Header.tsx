import { StyledLink } from "../../styles/StyledLink";
import { HeaderStyled } from "../../styles/HeaderStyled";
import { useContext } from "react";
import { CartProductsContext } from "../../ProductsContext/CartProductsContext";

export const Header = () => {
  const { cartProducts } = useContext(CartProductsContext);

  let cartCount = 0;

  cartProducts.forEach((product) => {
    cartCount += product.amount;
  });

  return (
    <HeaderStyled>
      <StyledLink to="/">Home</StyledLink>
      <StyledLink to="/cart">Cart {cartCount}</StyledLink>
    </HeaderStyled>
  );
};
