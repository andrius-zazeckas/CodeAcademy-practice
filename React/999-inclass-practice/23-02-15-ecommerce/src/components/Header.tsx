import { StyledLink } from "../styles/StyledLink";
import { HeaderStyled } from "../styles/HeaderStyled";

export const Header = () => {
  return (
    <HeaderStyled>
      <StyledLink to="/">Home</StyledLink>
      <StyledLink to="/cart">Cart</StyledLink>
    </HeaderStyled>
  );
};
