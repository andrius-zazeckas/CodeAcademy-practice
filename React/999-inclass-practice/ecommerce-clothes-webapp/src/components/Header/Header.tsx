import { Box, Grid, Typography } from "@mui/material";
import type { FC } from "react";
import { Link, useLocation } from "react-router-dom";

export const Header: FC = () => {
  const { pathname } = useLocation();
  const isOnCartLink = pathname.includes("/cart");

  return (
    <Box
      component="header"
      textAlign="center"
      margin="0 auto"
      // width="100"
      borderBottom="2px solid black"
    >
      <Typography variant="h3" padding={2} fontWeight="600" fontSize="44px">
        {isOnCartLink ? "CART" : "PRODUCTS"}
      </Typography>

      {/* <div role="products"></div> */}

      {/* TODO heading Products pagal link, sutvarkyti warning */}

      <Grid
        role="navigation"
        container
        textAlign="center"
        mb={2}
        sx={{
          "& a": {
            color: "purple",
            textDecoration: "none",

            ":hover": { color: "lightgreen" },
          },
        }}
      >
        <Grid item xs={12} sm={6}>
          <Link to="/">
            <Typography aria-label="home link" fontSize="32px">
              Home
            </Typography>
          </Link>
        </Grid>

        <Grid item xs={12} sm={6}>
          <Link to="/cart">
            <Typography aria-label="cart link" fontSize="32px">
              Cart
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </Box>
  );
};
