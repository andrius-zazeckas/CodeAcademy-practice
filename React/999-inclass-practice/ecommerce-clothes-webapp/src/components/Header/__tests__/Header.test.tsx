import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Header } from "../Header";

describe("Header", () => {
  //   it("should render links", () => {
  //     render(
  //       <BrowserRouter>
  //         <Header />
  //       </BrowserRouter>
  //     );

  //     expect(screen.getByLabelText("home link")).toBeVisible();
  //     expect(screen.getByLabelText("cart link")).toBeVisible();
  //   });

  it.each(["home link", "cart link"])("should render %s", (link) => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(screen.getByLabelText(link)).toBeVisible();
  });
});
