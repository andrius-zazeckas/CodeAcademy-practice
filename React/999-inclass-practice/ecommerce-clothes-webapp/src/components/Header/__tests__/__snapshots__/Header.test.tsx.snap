// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  className="MuiBox-root css-wyu7nf"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 css-93metl-MuiTypography-root"
  >
    PRODUCTS
  </h3>
  <div
    className="MuiGrid-root MuiGrid-container css-16gs291-MuiGrid-root"
    role="navigation"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <p
          aria-label="home link"
          className="MuiTypography-root MuiTypography-body1 css-fldieq-MuiTypography-root"
        >
          Home
        </p>
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
    >
      <a
        href="/cart"
        onClick={[Function]}
      >
        <p
          aria-label="cart link"
          className="MuiTypography-root MuiTypography-body1 css-fldieq-MuiTypography-root"
        >
          Cart
        </p>
      </a>
    </div>
  </div>
</header>
`;
